spec/facility_spec.rb

```diff
@@ -23,4 +29,173 @@
+      @facility.register_vehicle(cruz)
+
+      expect(@facility.vehicle_registered?(cruz)).to eq(false)
+    end
+
+    it 'will store a new plate type once the vehicle is registered' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(cruz)
+
+      expect(cruz.plate_type).to eq(:regular)
+    end
+
+    it 'will store the date the vehicle was registerd on' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(cruz)
+
+      expect(cruz.registration_date).to eq(Date.today)
+    end
+  end
+
+
+  describe '#add_service' do
+    it 'can add a service' do
+      @facility.add_service('Written Test')
+
+      expect(@facility.services).to eq(['Written Test'])
+    end
+  end
+
+  describe '#cost' do
+  it 'can collect fees' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(cruz)
+
+      expect(@facility.collected_fees).to be(100)
+    end
+  
+
+  it 'can keep count of total fees collected' do
+    cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+    bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+    @facility.add_service('Vehicle Registration')
+    @facility.register_vehicle(cruz)
+    @facility.register_vehicle(bolt)
+
+    expect(@facility.collected_fees).to be(300)
+    # binding.pry
+    end
+  end
+
+  describe '#administer_written_test' do
+    it 'can assign a written test to a registrant' do
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      
+      @facility.add_service('Written Test')
+      @facility.administer_written_test(registrant_1)
+
+      expect(registrant_1.license_data[:written]).to eq(true)
+    end
+
+    it 'will not assign a written test if facility does not offer written tests' do
+      registrant_1 = Registrant.new('Bruce', 18, true )
+      
+      @facility.administer_written_test(registrant_1)
+  
+      expect(registrant_1.license_data[:written]).to eq(false)
+      end
+
+      it 'will not assign a written test if registrant does not have a permit' do
+        registrant_2 = Registrant.new('Penny', 16)
+        
+        @facility.administer_written_test(registrant_2)
+    
+        expect(registrant_2.license_data[:written]).to eq(false)
+        end
+
+    it 'will not assign a written test to anyone under the age of 16' do
+      registrant_3 = Registrant.new('Tucker', 15, true )
+      
+      @facility.administer_written_test(registrant_3)
+  
+      expect(registrant_3.license_data[:written]).to eq(false)
+      end
+  end
```

Comment:
Great job here testing all the combinations of conditions to test all the ways this method could function.